default/ship_parts.txt

```diff
@@ -1134,11 +1134,22 @@ Count condition = And [
             ]
 '''
 
+// Beginning of Stacking macros
+/* For the AI and general uniformity, shields, detectors and cloaks are assigned a capacity.
+However, to prevent stacking, for each group X there is a X_STACKING macro which basically 
+first removes all effects of the used parts (by subtracting the summed capacity of all used parts)
```

Comment:
I'd say this entire thing of complicated scripting to get reasonable stacking behavior working shows that the stacking code on the C++ side needs an overhaul. There needs to be a way to specify in the scripts more precisely how certain effects should stack.

The most simple approach would be to be able to specify a "stacking priority" alongside the stacking group in an effectsgroup. The effectsgroup with the highest stacking priority within a stacking group will be applied.